.github/workflows/tests.yml

```diff
@@ -151,52 +210,41 @@ jobs:
           draft: true
           delete-branch: true
 
-      # return failure if any checks failed
-      # return failure if any checks were cancelled
-      # loop again if any checks are still in progress
-      # loop again if no checks have completed yet
-      # return failure if checks are unchanged since last loop and the checks are empty
-      # return success if checks are unchanged since last loop and at least once check is completed
-      # (else) loop again if checks have changed since last loop (avoid race conditions)
+      # wait for checks register, complete and fail if any required checks failed
       - name: Wait for all checks to finish
         if: steps.cpr.outputs.pull-request-number != ''
         env:
           flowzone_re: "^Flowzone"
+          filter_re: "^policy-?bot"
+          GH_TOKEN: ${{ steps.gh_token.outputs.token || secrets.FLOWZONE_TOKEN }}
         run: |
-          while true
-          do
-            sleep 30
-
-            prev_checks="${checks}"
-
-            checks="$(gh pr status --json statusCheckRollup,number --jq '[.createdBy[] |
```

Comment:
`.createdBy` is empty when queried using ephemeral tokens. Must be a GH permissions thing requiring a token tied to a user (not machine/app) identity. No way around it as far as I can tell. I've re-implemented using `gh pr checks`, which is a bit kludgy but seems to work).

Reply:
I like how much cleaner it is, but I think `gh pr checks --required` will never pass if we change the name of a job.

These draft PRs intentionally don't update branch protections, so renamed jobs will be pending forever.

Reply:
@klutchell you'll notice that we always filter the output and don't use the exit code from GH CLI.